index.js

```diff
@@ -23,6 +24,7 @@ class bot extends core {
         this.config_handler =  new config.config_handler(config_file_path); // Initalise a new object with the config file
         this.config_handler.load(true); // Load the config
         this.config = this.config_handler.config; // Set a shorter variable name since accessing it is easier now
+        new caps(this, config.caps).run()

```

Comment:
Could this be changed into this.caps = caps(this) so Caps get's the class property from this instead of having an extra argument? Also .run() is a nice idea but I'd rather have Caps being run when it's being initialised in index.js.

Reply:
I can change it to only require one argument: `this`, as for the rest I need more clarifications